tests/handlers/base/test_opt.py

```diff
@@ -37,3 +48,85 @@ def test_opt_settings(decorator: "RouteHandlerType", handler: Callable) -> None:
     base_opt = {"base": 1, "kwarg_value": 0}
     result = decorator("/", opt=base_opt, kwarg_value=2)(handler)  # type: ignore
     assert result.opt == {"base": 1, "kwarg_value": 2}
+
+
+@pytest.mark.parametrize(
+    "app_opt, router_opt, controller_opt, route_opt, expected_opt",
+    [
+        [
+            {"app": "app"},
+            {"router": "router"},
+            {"controller": "controller"},
+            {"route": "route"},
+            {"app": "app", "router": "router", "controller": "controller", "route": "route"},
+        ],
+        [
+            {"override": "app"},
+            {"router": "router"},
+            None,
+            {"override": "route"},
+            {"router": "router", "override": "route"},
+        ],
+        [None, None, None, None, {}],
+    ],
+)
+def test_opt_resolution(
+    app_opt: Dict[str, Any],
+    router_opt: Dict[str, Any],
+    controller_opt: Dict[str, Any],
+    route_opt: Dict[str, Any],
+    expected_opt: Dict[str, Any],
+) -> None:
+    class MyController(Controller):
+        path = "/controller"
+        opt = controller_opt
+
+        @get(opt=route_opt)
+        def handler(self) -> None:
+            ...
+
+    router = Router("/router", route_handlers=[MyController], opt=router_opt)
+    app = Starlite(route_handlers=[router], opt=app_opt)
+    assert (
+        app.asgi_router.root_route_map_node["children"]["/router/controller"]["asgi_handlers"]["GET"][1].opt
+        == expected_opt
+    )
+
+
+def test_opt_not_affected_by_route_handler_copying() -> None:
```

Comment:
I've added this test mostly because copying handlers from a controller contains complex logic.